crates/biome_js_formatter/src/context.rs

```diff
@@ -531,3 +550,47 @@ impl fmt::Display for ArrowParentheses {
         }
     }
 }
+
+#[derive(Clone, Copy, Debug, Default, Deserializable, Eq, Hash, Merge, PartialEq)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize),
+    serde(rename_all = "camelCase")
+)]
+#[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
+pub enum ObjectWrap {
+    #[default]
+    Preserve,
+    Collapse,
+}
+
+impl ObjectWrap {
+    pub const fn is_preserve(&self) -> bool {
+        matches!(self, Self::Preserve)
+    }
+
+    pub const fn is_collapse(&self) -> bool {
+        matches!(self, Self::Collapse)
+    }
+}
+
+impl FromStr for ObjectWrap {
+    type Err = &'static str;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "as-needed"  => Ok(Self::Preserve),
+            "always"  => Ok(Self::Collapse),
```

Comment:
This doesn't look right.

```suggestion
            "preserve"  => Ok(Self::Preserve),
            "collapse"  => Ok(Self::Collapse),
```

Reply:
My bad, thank you for catching this. Fixed in [5eb499b](https://github.com/biomejs/biome/pull/5068/commits/5eb499b376c4c543193f5db2d4f4f837702b935a)